src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -52,32 +54,39 @@
     value = "RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE",
     justification = "Spotbugs doesn't understand resource try construct")
 class FlightDao {
-  private static final Logger logger = LoggerFactory.getLogger(FlightDao.class);
-
   static final String FLIGHT_TABLE = "flight";
   static final String FLIGHT_LOG_TABLE = "flightlog";
   static final String FLIGHT_INPUT_TABLE = "flightinput";
   static final String FLIGHT_WORKING_TABLE = "flightworking";
-
+  private static final Logger logger = LoggerFactory.getLogger(FlightDao.class);
   private static final String UNKNOWN = "<unknown>";
 
   private final DataSource dataSource;
   private final StairwayInstanceDao stairwayInstanceDao;
   private final ExceptionSerializer exceptionSerializer;
-  private final boolean keepFlightLog;
   private final HookWrapper hookWrapper;
+  private final int flightVisibilitySeconds;
 
   FlightDao(
       DataSource dataSource,
       StairwayInstanceDao stairwayInstanceDao,
       ExceptionSerializer exceptionSerializer,
       HookWrapper hookWrapper,
-      boolean keepFlightLog) {
+      Duration completedFlightAvailable) {
     this.dataSource = dataSource;
     this.stairwayInstanceDao = stairwayInstanceDao;
     this.exceptionSerializer = exceptionSerializer;
     this.hookWrapper = hookWrapper;
-    this.keepFlightLog = keepFlightLog;
+
+    int visibility = 0;
+    if (completedFlightAvailable != null) {
```

Comment:
Is the intention that a null `completedFlightAvailable` implies "keep forever"?  If so it might be good to make this more explicit via doc/`@nullable`/`Optional<>` (no strong opinion on how).